reactive/etcd.py

```diff
@@ -821,6 +834,78 @@ def remove_nrpe_config(nagios=None):
         nrpe_setup.remove_check(shortname=service)
 
 
+@when('endpoint.prometheus.joined',
+      'leadership.is_leader',
+      'certificates.ca.available')
+def register_prometheus_jobs():
+    log('Registering Prometheus metrics collection.')
+    prometheus = endpoint_from_flag('endpoint.prometheus.joined')
+    cluster = endpoint_from_flag('cluster.joined')
+
+    if not etcd_reachable_from_endpoint(prometheus.endpoint_name):
+        log('Aborting Prometheus metrics collection. Etcd is not reachable by'
+            ' prometheus client', WARNING)
+        return
+
+    peer_ips = cluster.get_db_ingress_addresses() if cluster else []
+    peer_ips.append(get_ingress_address('db'))
+    targets = ["{}:{}".format(ip, config('port')) for ip in peer_ips]
+    log('Configuring Prometheus scrape targets: {}'.format(targets), DEBUG)
+    prometheus.register_job(job_name='etcd',
+                            job_data={
+                                'scheme': 'https',
+                                'static_configs': [
+                                    {'targets': targets},
+                                ]
+                            })
+    set_flag('prometheus.configured')
+
+
+@when('endpoint.prometheus.departed')
+def reset_prometheus_config():
+    clear_flag('prometheus.configured')
+    clear_flag('grafana.configured')
```

Comment:
This isn't the flag you want.  It's set whenever any single unit goes away and is never automatically cleared (see [docs](https://charmsreactive.readthedocs.io/en/latest/charms.reactive.relations.html#charms.reactive.endpoints.Endpoint)).  You want `@when_not('endpoint.prometheus.joined')` or even better, use [`register_trigger()`](https://charmsreactive.readthedocs.io/en/latest/charms.reactive.flags.html#charms.reactive.flags.register_trigger) since it's guaranteed to run before decorated handlers (like `register_grafana_dashboard`, below):

```suggestion
register_trigger(when_not="endpoint.prometheus.joined", clear_flag="prometheus.configured")
register_trigger(when_not="endpoint.prometheus.joined", clear_flag="grafana.configured")
```